backend/routes/auth.js

```diff
@@ -0,0 +1,75 @@
+const { PrismaClient } = require('../generated/prisma')
+const prisma = new PrismaClient()
+const router = require('express').Router()
+const bcrypt = require("bcrypt");
+
+router.post('/auth/signup', async(req, res) => {
+    const { username, password, name, role } = req.body
+    if(!username || !password) {
+        return res.status(400).json({ error: "Username and password are required." })
+    }
+    
+    try {
+        const existingUser = await prisma.user.findUnique({
+            where: { username }
+        })
+        if(existingUser) {
+            return res.status(400).json({ error: "Username already taken." });
+        }
+
+        const hashedPassword = await bcrypt.hash(password, 10)
+        const newUser = await prisma.user.create({
+            data: { username, password: hashedPassword, name, role }
+        })
+        res.status(201).json({ message: "User created successfully!" });
+    } catch(error) {
+        console.log(error);  
+        res.status(500).json({ error: 'Server error' });
+    }
+})
+
+router.post('/auth/login', async(req, res) => {
+    const { username, password } = req.body
+    if(!username || !password) {
+        return res.status(400).json({ error: "Username and password are required." })
+    }
+
+    try {
+        const user = await prisma.user.findUnique({
+            where: { username }
+        })
+        if(!user) {
+            return res.status(400).json({ error: "Invalid username or password." })
+        }
+
+        const isValidPassword = await bcrypt.compare(password, user.password)
+        if(!isValidPassword) {
+            return res.status(400).json({ error: "Invalid username or password." })
+        }
+
+        req.session.userId = user.id;
+        res.json({ message: "Login successful!" })
+    } catch(error) {
+        console.log(error);  
+        res.status(500).json({ error: 'Server error' });
+    }
+})
+
+router.post('/auth/me', async(req, res) => {
```

Comment:
Is this a temporary endpoint? (I don't see it outlined in the Capstone Proposal and it seems to only print basic info about who is signed in.) If so, I would recommend making a comment or note about this endpoint.